### PR TITLE
Added support for .tmx maps

### PR DESCRIPTION
**Pull request by Nanogamer7**

### CODE REVIEW

File: src/Map.cpp [hunk 1]
```diff
@@ -63,4 +78,93 @@ void Map::addTile(unsigned long id, int x, int y, GameInternal* game, const std:
...
+            auto& tileLayer = layer->getLayerAs<tmx::TileLayer>();
+
+            int zIndex = 0;
+
+            const std::vector<tmx::Property>& properties = layer->getProperties();
+            auto zIndexIterator = std::find_if(properties.begin(), properties.end(), [](const tmx::Property& property) {
+                return property.getName() == "zIndex";
+            });
+
+            if (zIndexIterator != properties.end() && std::is_same<decltype(zIndexIterator->getType()), int>::value) {
```
Comment (Freezarite) [inline src/Map.cpp:114]: Should be castable, is_same is too specific

Comment (Freezarite) [inline src/Map.cpp]: Should be castable, is_same is too specific

File: src/Map.cpp [hunk 2]
```diff
@@ -63,4 +78,93 @@ void Map::addTile(unsigned long id, int x, int y, GameInternal* game, const std:
...
+                zIndex = zIndexIterator->getIntValue();
+            }
+
+            const auto& tiles = tileLayer.getTiles();
+            
+            for (auto i = 0u; i < tileSets.size(); i++) {
+                auto tilesetTexture = VEGO_Game().textureManager->loadTexture(texturePaths.at(i).c_str());
```
Comment (Freezarite) [inline src/Map.cpp:122]: Does not work with upcoming texturemanager changes

File: src/Map.cpp [hunk 3]
```diff
@@ -63,4 +78,93 @@ void Map::addTile(unsigned long id, int x, int y, GameInternal* game, const std:
...
+                int texX, texY;
```
Comment (Freezarite) [inline src/Map.cpp]: maybe better variable names

File: src/Map.cpp [hunk 4]
```diff
@@ -63,4 +78,93 @@ void Map::addTile(unsigned long id, int x, int y, GameInternal* game, const std:
...
+                SDL_QueryTexture(tilesetTexture, nullptr, nullptr, &texX, &texY);
+
+                const auto tileCountX = texX / mapTileSize.x;
+                const auto tileCountY = texY / mapTileSize.y;
+
+                for (auto idx = 0ul; idx < mapSize.x * mapSize.y; idx++) {
+
+                    if (!(idx < tiles.size() && tiles[idx].ID >= tileSets.at(i).getFirstGID()
+                        && tiles[idx].ID < (tileSets.at(i).getFirstGID() + tileSets.at(i).getTileCount()))) {
```
Comment (Freezarite) [inline src/Map.cpp]: not statement for exit/continue conditions makes it harder to read if they are not necessary

**Review by Freezarite — CHANGES_REQUESTED** [review]

File: src/Map.cpp (hunk 2 above)
Comment (Nanogamer7) [reply src/Map.cpp:122]: Will be resolved on merge of #86

**Review by Freezarite — APPROVED** [review]
lgtm

**Comment by Nanogamer7** [comment]
Closes #53

**Pull request merged**